### PR TITLE
Some more cleanup in abstract classes in preparation of splitting

### PR DESCRIPTION
**Pull request by akashkulk**

### CODE REVIEW

**Review by rodireich — APPROVED** [review]

File: airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java [hunk 1]
```diff
@@ -4,23 +4,66 @@
...
 import java.util.Objects;
 import java.util.function.Function;
 import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+/**
+ * Contains utilities and helper classes for discovering schemas in database sources.
+ */
 public class DbSourceDiscoverUtil {
```
Comment (nguyenaiden) [inline airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java:32]: I'm good with leaving testing of this class to blackbox at the connector level for now. But we should unit test this at some point. 
Also, making these methods static is appropriate here, but I always have run into an issue trying to test static methods.

**Review by nguyenaiden — APPROVED** [review]
+1 with a small comment.

File: airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java [hunk 2]
```diff
@@ -4,23 +4,66 @@
...
+          .stream()
+          .map(commonField -> toField(commonField, airbyteTypeConverter))
+          .distinct()
+          .collect(Collectors.toList());
+      final JsonNode currentJsonSchema = fieldsToJsonSchema(fields);
+
+      final JsonNode catalogSchema = stream.getJsonSchema();
+      if (!catalogSchema.equals(currentJsonSchema)) {
+        LOGGER.warn(
+            "Source schema changed for table  {}! Actual schema: {}. Catalog schema:  {}",
```
Comment (nguyenaiden) [inline airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/DbSourceDiscoverUtil.java:59]: nit: extra space between `table  {}` and `Catalog schema:  {}`

**Pull request merged**